pkg/controllers/vdb/init_db.go

```diff
@@ -514,85 +462,81 @@ func (g *GenericDatabaseInitializer) getEnableHTTPS() string {
 	return "0"
 }
 
-// getCAFile will return an entry for SystemCABundlePath if one needs to be included
-func (g *GenericDatabaseInitializer) getCAFile() string {
-	if g.Vdb.Spec.Communal.CaFile == "" {
-		return ""
+// setCAFile adds an entry for SystemCABundlePath, if one needs to be included,
+// to the config parms map.
+func (g *GenericDatabaseInitializer) setCAFile() {
+	if g.Vdb.Spec.Communal.CaFile != "" {
+		g.ConfigurationParams.Set("SystemCABundlePath", g.Vdb.Spec.Communal.CaFile)
 	}
-	return fmt.Sprintf("SystemCABundlePath = %s", g.Vdb.Spec.Communal.CaFile)
 }
 
-// getServerSideEncryptionParmsContent constructs a string for server-side encryption related auth
-// parms if that is setup.  Must have encryption type in the Vdb.
-func (g *GenericDatabaseInitializer) getServerSideEncryptionParmsContent(ctx context.Context) (string, reconcile.Result, error) {
+// setServerSideEncryptionParms adds server-side encryption related config
+// parms, if that is setup, to the config parms map. Must have encryption type in the Vdb.
+func (g *GenericDatabaseInitializer) setServerSideEncryptionParms(ctx context.Context) (reconcile.Result, error) {
 	// Extract customer key from secret
-	clienKey, res, err := g.getS3SseCustomerKey(ctx)
+	res, err := g.setS3SseCustomerKey(ctx)
 	if verrors.IsReconcileAborted(res, err) {
-		return "", res, err
+		return res, err
 	}
 
-	content := fmt.Sprintf(`
-			%s
-			%s
-			%s
-		`, g.getServerSideEncryptionAlgorithm(), g.getS3SseKmsKeyID(), clienKey,
-	)
-	return content, ctrl.Result{}, nil
+	g.setServerSideEncryptionAlgorithm()
+	g.setS3SseKmsKeyID()
+	return ctrl.Result{}, nil
 }
 
-// getServerSideEncryptionAlgorithm returns an entry for S3ServerSideEncryption
-// if sse type is SSE-S3|SSE-KMS or for S3SseCustomerAlgorithm if SSE-C.
-func (g *GenericDatabaseInitializer) getServerSideEncryptionAlgorithm() string {
+// setServerSideEncryptionAlgorithm adds an entry to the config parms map for S3ServerSideEncryption,
+// if sse type is SSE-S3|SSE-KMS, or for S3SseCustomerAlgorithm, if sse type is SSE-C.
+func (g *GenericDatabaseInitializer) setServerSideEncryptionAlgorithm() {
 	if g.Vdb.IsSseC() {
```

Comment:
Can we change the `if {} if {} if {}` calls to a `switch` statement? Before we use to return once the first one succeeded. But now, we will check the next if.